kolibri/core/assets/src/vue/core-modal/index.vue

```diff
@@ -109,6 +102,9 @@
       closeModal() {
         this.$dispatch('close');
       },
+      submitModal() {
+        this.$dispatch('submit');
+      },
```

Comment:
I actually wanted to touch base with you on how events are used by the modals. Right now it seems like they're both emitting and listening to them at the same time, which I didn't quite understand.

If you'd like to get a sense for how I've been thinking about events and modals, you can see my "fake modal" in use here: https://github.com/learningequality/kolibri/blob/master/kolibri/plugins/management/assets/src/vue/content-page/wizard-import-local.vue

Basically, I give the `modal` component events like `cancel` and `submit`, and the specific modal component (in this case, the `wizard-import-local` component) can listen to these events. The events are triggered based on user actions: hitting 'escape', clicking the background, or clicking the 'x' all trigger a `cancel` event; hitting 'enter' triggers a `submit` event.

These are analogous to how a form might have a `submit` event or a link has a `click` event. They don't indicate that the form was actually submitted or a link was followed: instead they indicate that the user made some action...

So the event _listener_ (the wizard form) decides whether to actually submit or cancel. The event _emitter_ (the generic `modal`) just indicates that some user input occurred, and nothing more.

This strategy doesn't always work - the main exception seems to be the `open` event which indicates that the modal WAS OPENED, not that the user requested it to be opened. However, I'm wondering whether the `open` and `close` events might be replaced with Vue's built-in `attached` and `detached` events?